sisl/grid.py

```diff
@@ -179,6 +179,84 @@ def interp(self, shape, order=1, mode="wrap", **kwargs):
 
         # Apply the scipy.ndimage.zoom function and return a new grid
         return self.apply(ndimage_zoom, zoom_factors, mode=mode, order=order, **kwargs)
+    
+    def transform_axes(self, axes=np.eye(3), output_shape=None, mode="constant", order=1, **kwargs):
+        """
+        Applies a linear transformation to the grid to get it relative to arbitrary axes.
+
+        This method can be used, for example to get the values of the grid with respect to
+        the standard basis, so that you can easily visualize it or overlap it with other grids
+        (e.g. to perform integrals).
+
+        Parameters
+        -----------
+        axes: array-like of shape (3,3)
+            these axes represent the directions that you want to use as references for
+            the new grid.
+
+            The length of the axes does not have any effect! They will be rescaled to create
+            the minimum bounding box necessary to accomodate the unit cell.
+        output_shape: array-like of int of shape (3,), optional
+            the shape of the final output. If not provided, the current shape of the grid
+            will be used. 
+            
+            Notice however that if the transformation applies a big shear to the image (grid)
+            you will probably need to have a bigger output_shape.
+        mode: str, optional
+            determines how to handle borders. See scipy docs for more info on the possible values.
+        order : int 0-5, optional
+            the order of the spline interpolation to calculate the values (since we are applying
+            a transformation, we don't actually have values for the new locations and we need to
+            interpolate them)
+            1 means linear, 2 quadratic, etc...
+        **kwargs:
+            the rest of keyword arguments are passed directly to `scipy.ndimage.affine_transform`
+
+        See also
+        ----------
+        scipy.ndimage.affine_transform : method used to apply the linear transformation.
+        """
+        # Take the current shape of the grid if no output shape was provided
+        if output_shape is None:
+            output_shape = self.shape
+
+        # Get the current cell in coordinates of the destination axes
+        inv_axes = np.linalg.inv(axes)
```

Comment:
this one is inherent to `SuperCell` so once that is used this is easier :)

Reply:
Although I still need to do `icell = sc.icell` right? Otherwise it will recalculate the inverse each time I do `sc.icell`.

Reply:
I noticed that `cell_invert` is 7 times faster than `np.linalg.inv`, but instancing a `SuperCell` adds a lot of overhead:

```python
sinvert = sisl._supercell.cell_invert
npinvert = np.linalg.inv

%timeit npinvert(cell)
%timeit sinvert(cell)
%timeit sisl.SuperCell(cell).icell
```
_7.36 µs ± 66.2 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)
1.06 µs ± 11.1 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)
48.4 µs ± 142 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)_

Should I use `cell_invert` directly and don't accept supercells? :thinking:

Reply:
No, if you compare with the scipy call this part is nothing ;)

So prefer readability and usability in this case.

Reply:
Right :sweat_smile: